client/checkout/classic/payment-processing.js

```diff
@@ -214,13 +214,20 @@ async function createStripePaymentElement(
 ) {
 	const amount = Number( getUPEConfig( 'cartTotal' ) );
 	const paymentMethodTypes = getPaymentMethodTypes( paymentMethodType );
+	const appearance = await initializeAppearance( api, elementsLocation );
+	document
+		.querySelector(
+			`.wcpay-upe-element[data-payment-method-type="${ paymentMethodType }"]`
+		)
+		?.closest( '.wc_payment_method' )
+		?.classList.add( `theme--${ appearance.theme || 'stripe' }` );
```

Comment:
This was a bit unfortunate - having to navigate the DOM. But I guess that's the shortcode checkout way of things.

It's worth mentioning that the `.theme--{themeName}` class name isn't applied to the same element on blocks and shortcode. But it doesn't really matter - it just needs to be applied somewhere in the hierarchy.
The structure of blocks and shortcode checkout is different, anyways - so it's be hard to be consistent.